research/conditional/moe_layers/expert_choice.py

```diff
@@ -27,6 +27,8 @@ def __init__(
         use_full_einsum: bool = False,
         softmax_over: Literal["tokens", "experts"] = "tokens",
         n_gating_heatmaps: int = 4,
+        init_type: Literal["switch", "regular"] = "switch",
+        init_scale: float = 0.1,
```

Comment:
to powinno być switch_init_scale, jeśli ma działać wyłącznie jak init_type jest równy switch. Ale jeszcze lepiej to wgl usunąć, imho, init_type, zostawić tylko init_scale z wartością domyślną 0.1, i ustawiać na 1.0 jeśli to ma być regular czy coś.

Jeśli już koniecznie chcemy flagę init_type z wartościami switch/regular, to imho to powinno być poza tą klasą. Czyli flaga jest init_type, ale robimy if'a przy parsowaniu argumentów i odpowiednio ustawiamy init_scale tutaj.

Ponadto, wgl, czy swtich nie skaluje _wszystkich_ inicjalizacji, a nie FF?